src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java

```diff
@@ -228,18 +228,22 @@ public Custody getCustodyByConvictionId(final String crn, final Long convictionI
     }
 
     @Transactional
-    public Custody offenderRecalled(final String nomsNumber, final LocalDate occurred) {
+    public Custody offenderRecalled(final String nomsNumber, final OffenderRecalledNotification recalledNotification) {
         final var offender = offenderRepository.findByNomsNumber(nomsNumber)
             .orElseThrow(() -> new NotFoundException(String.format("Offender with nomsNumber %s not found", nomsNumber)));
 
-        log.info("Offender {} recalled on {}", nomsNumber, occurred);
+        log.info("Offender {} recalled from {} on {}", nomsNumber, recalledNotification.getNomsPrisonInstitutionCode(), recalledNotification.getRecallDate());
+        final var telemetryProperties = Map.of("offenderNo", nomsNumber,
+            "recallDate", recalledNotification.getRecallDate().format(DateTimeFormatter.ISO_DATE),
+            "institution", recalledNotification.getNomsPrisonInstitutionCode());
 
         try {
             Event event = convictionService.getActiveCustodialEvent(offender.getOffenderId());
-            telemetryClient.trackEvent( "P2POffenderRecalled");
+
+            telemetryClient.trackEvent("P2POffenderRecalled", telemetryProperties, null);
             return ConvictionTransformer.custodyOf(event.getDisposal().getCustody());
```

Comment:
new field only looks like it is used in logging?

Reply:
Yes, it's a placeholder until the Delius team let us know what they want.